elements/modal.js

```diff
@@ -0,0 +1,68 @@
+const widget = require('cache-element/widget')
+const Modal = require('../lib/modal-element')
+const html = require('choo/html')
+
+module.exports = createWidget
+
+function createWidget () {
+  var link = null
+
+  var modal = Modal(html`<div></div>`, function () {
+    link = null
+  })
+
+  return widget({
+    onload: function () {
+      if (link) modal.show(createHtml(link))
+    },
+    onunload: function () {
+      link = null
+      modal.hide()
+    },
+    onupdate: function (el, linkArg) {
```

Comment:
i think this would be cleaner, and not even need the comments, with a different name for `linkArg`, like `newLink`